C/tests/c4CollectionTest.cc

```diff
@@ -190,12 +190,25 @@ N_WAY_TEST_CASE_METHOD(C4CollectionTest, "Collection Lifecycle Multi-DB", "[Data
     C4Database* db2 = c4db_openAgain(db, ERROR_INFO());
     REQUIRE(db2);
 
+
     Retained<C4Collection> guitars = c4db_createCollection(db, Guitars, ERROR_INFO());
     REQUIRE(guitars);
 
+    {
+        C4Database::Transaction t(db);
+        addNumberedDocs(guitars, 100);
+        t.commit();
+    }
+
     Retained<C4Collection> guitars2 = c4db_getCollection(db2, Guitars, ERROR_INFO());
     REQUIRE(guitars2);
 
+    CHECK(c4coll_getDocumentCount(guitars) == 100);
+    CHECK(c4coll_getDocumentCount(guitars2) == 100);
+    CHECK(c4coll_getLastSequence(guitars) == 100_seq);
+    CHECK(c4coll_getLastSequence(guitars2) == 100_seq);
+
+    // First delete the collection on the second DB instances
     REQUIRE(c4db_deleteCollection(db2, Guitars, ERROR_INFO()));
```

Comment:
The asymmetry between guitars and guitars2 as regard to Retained: "guitars2" must be retained or the next statement will blow up, but "guitars" does not have to be Retained only because it is the not derived from db2 by which the collection is deleted. It is awkward to describe. I wonder if you would consider to also keep it in the _collections for the db instance that has the collection be deleted? It won't harm to keep it in _collections as we already have the code to deal with it.

Reply:
I don't consider this essential since we have a clear warning about this in the header, but on the other hand "why not?" comes to mind.